.github/workflows/.reusable-frontend-deploy.yml

```diff
@@ -0,0 +1,50 @@
+# reusable workflow
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        description: Github environment to use for secret / variable access
+        required: true
+
+defaults:
+  run:
+    working-directory: frontend
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Cloning repo
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 16.x
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: npm-
+
+      - name: Deploy to Vercel
+        env:
+          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+          ENV: prod
```

Comment:
I think it might be worth mentioning this is a production workflow in the filename if this isn't an input

Reply:
Yeah, fair point. I've added an input for this as I hope that we can use this for staging in the future rather than having different workflows for staging / production and demo.